src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -20,54 +18,37 @@
 using Raven.Server.ServerWide;
 using Raven.Server.ServerWide.Commands.Subscriptions;
 using Raven.Server.ServerWide.Context;
-using Raven.Server.Utils;
-using Sparrow.Binary;
 using Sparrow.Logging;
-using Sparrow.LowMemory;
-using Voron;
 
 namespace Raven.Server.Documents.Subscriptions
 {
-    public partial class SubscriptionStorage : IDisposable, ILowMemoryHandler, ISubscriptionSemaphore
+    public class SubscriptionStorage : AbstractSubscriptionStorage<SubscriptionConnectionsState>
```

Comment:
is it possible to do 
`public class SubscriptionStorage<T> : AbstractSubscriptionStorage<T> where T : SubscriptionConnectionsState`

so you will be able to have `SubscriptionStorage : SubscriptionStorage<SubscriptionConnectionsState>`
and 
`ShardSubscriptionStorage : SubscriptionStorage<ShardSubscriptionConnectionsState>`

But it might be over engineering :)